### PR TITLE
Update KIcon/KLabeledIcon API, Simplify KLabeledIcon most-common usage

### LINKED ISSUE #5714: change KIcon API
### Observed behavior

see poll: https://learningequality.slack.com/archives/CB37UM23A/p1556059923121200

The current `KIcon` API looks like

```html
<KIcon back />
```

### Expected behavior

This should be updated to 

```html
<KIcon icon="back" />
```

### User-facing consequences

None

### Context

0.12.4

### PR DESCRIPTION
**Pull request by jonboiser**
### Summary

1. Adds an 'icon' prop for KIcon, removes the boolean props. Fixes #5714 
1. Adds 'icon' and 'label' props for KLabeledIcon to simplify the most-common usage of KLabeledIcon, and simplify implementation of #5729 (won't have to add a lot of extra `template` tags). Also centralized control of the `dir` attribute to KLabeledIcon when the 'label' prop is provided to it.

### Reviewer guidance

1. Look at each change individually. Make sure 'label' and 'icon' props were properly added/updated.
1. Browse through the Learn and Coach pages to see if any icons disappeared.

### References

…

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/core/assets/src/views/icons/KBasicContentIcon.vue [hunk 1]
```diff
@@ -1,13 +1,6 @@
...
 
-  <KIcon
-    slot="icon"
-    :exercise="kind === 'exercise'"
-    :video="kind === 'video'"
-    :app="kind === 'html5'"
-    :audio="kind === 'audio'"
-    :doc="kind === 'document'"
-  />
+  <KIcon slot="icon" :icon="kind" />
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/icons/KBasicContentIcon.vue]: Looks like `KBasicContentIcon` can be entirely replaced by `KIcon` then?

File: kolibri/core/assets/src/views/icons/KIcon.vue [hunk 2]
```diff
@@ -53,15 +52,61 @@
...
   export default {
     name: 'KIcon',
     mixins: [themeMixin],
     props: {
+      icon: {
+        type: String,
+        required: true,
+        validator(value) {
+          return iconTypes.includes(value);
+        },
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/icons/KIcon.vue:109]: nice

File: kolibri/core/assets/src/views/icons/KIcon.vue [hunk 3]
```diff
@@ -53,15 +52,61 @@
 
   import themeMixin from 'kolibri.coreVue.mixins.themeMixin';
 
-  const boolean = {
-    type: Boolean,
-    default: false,
-  };
+  const iconTypes = [
+    'correct',
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/icons/KIcon.vue:57]: missed the ` // tracking` comment during copy-paste

File: kolibri/core/assets/src/views/icons/KLabeledIcon.vue [hunk 4]
```diff
@@ -2,12 +2,16 @@
 
   <span class="labeled-icon-wrapper">
     <div class="icon">
-      <slot name="icon"></slot>
+      <slot name="icon">
+        <KIcon v-if="icon" :icon="icon" />
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/icons/KLabeledIcon.vue:6]: nice

File: kolibri/core/assets/src/views/icons/KLabeledIcon.vue [hunk 5]
```diff
@@ -2,12 +2,16 @@
...
+      </slot>
     </div>
     <div class="label">
       <!-- nest slot inside span to get alignment and flow correct for mixed RLT/LTR -->
       <span dir="auto">
-        <slot></slot>
+        <slot>
+          {{ label }}
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/icons/KLabeledIcon.vue:13]: nice

File: kolibri/plugins/coach/assets/src/views/common/status/CoachStatusIcon.vue [hunk 6]
```diff
@@ -33,8 +27,13 @@
...
     computed: {
-      ICONS() {
-        return ICONS;
+      iconType() {
+        return {
+          [ICONS.clock]: 'inProgress',
+          [ICONS.star]: 'mastered',
+          [ICONS.help]: 'helpNeeded',
+          [ICONS.nothing]: 'notStarted',
+        }[this.icon];
```
Comment (indirectlylit) [inline kolibri/plugins/coach/assets/src/views/common/status/CoachStatusIcon.vue:36]: We should probably just get rid of `CoachStatusIcon` and these special-case `ICONS` constants, and them with `KIcon`? Could do this in a second pass to keep the PR simpler

File: kolibri/plugins/coach/assets/src/views/reports/ReportsExerciseLearners.vue [hunk 7]
```diff
@@ -15,14 +15,16 @@
...
+          <KLabeledIcon icon="person">
+            <KRouterLink
+              v-if="showLink(entry)"
+              :text="entry.name"
+              :to="entry.exerciseLearnerLink"
+            />
+            <template v-else>
+              {{ entry.name }}
+            </template>
+          </KLabeledIcon>
```
Comment (indirectlylit) [inline kolibri/plugins/coach/assets/src/views/reports/ReportsExerciseLearners.vue:27]: nice - good catch

File: kolibri/plugins/coach/assets/src/views/reports/ReportsLearnerReportLessonPage.vue [hunk 8]
```diff
@@ -31,12 +30,16 @@
...
+          <template slot="key">
+            {{ coachCommon$tr('descriptionLabel') }}
+          </template>
+          <template slot="value">
+            <span dir="auto">
+              {{ lesson.description || coachCommon$tr('descriptionMissingLabel') }}
+            </span>
+          </template>
         </HeaderTableRow>
-         -->
```
Comment (indirectlylit) [inline kolibri/plugins/coach/assets/src/views/reports/ReportsLearnerReportLessonPage.vue:39]: was this completely broken before?

**Review by indirectlylit — APPROVED** [review]
awesome!

File: kolibri/plugins/coach/assets/src/views/reports/ReportsLearnerReportLessonPage.vue (hunk 8 above)
Comment (jonboiser) [reply kolibri/plugins/coach/assets/src/views/reports/ReportsLearnerReportLessonPage.vue:39]: There are a lot of commented out sections like this in different report pages.

**Pull request merged**